File: java/ooad/TicTacToeApp/src/com/techlabs/tictactoe/Game.java
package com.techlabs.tictactoe;

public class Game {
	private Player player1;
	private Player player2;
	private Board board;
	private ResultAnalyzer resultAnalyzer;
	private Player currentPlayer;
	private boolean turn = true;
	private boolean winner = false;

	public Game(Player player1, Player player2, Board board,
			ResultAnalyzer resultAnalyzer) {
		this.player1 = player1;
		this.player2 = player2;
		this.board = board;
		this.resultAnalyzer = resultAnalyzer;
		this.currentPlayer=player1;
	}

	public void initializeGame() {
		this.board.displayBoard();
	}

	public void play(int position) {
		if (validChoice(position)) {
			this.currentPlayer=getTurn();
			this.board.updateBoard(this.getCurrentPlayer().getMark(), position);

			this.resultAnalyzer.analyzeBoard();

			if (turn) {
				this.turn = false;
			} else {
				this.turn = true;
			}
		}else{
			System.out.println("Choose a valid Option: "+this.getTurn());
		}
	}
	
	public Player getCurrentPlayer(){
		return this.currentPlayer;
	}
	public Board getBoard(){
		return this.board;
	}

	public boolean validChoice(int position) {
		if (!this.board.getCell(position-1).isMarked())
			return true;
		return false;
	}

	public Player getTurn() {
		if (turn) {
			return this.player1;
		} else {
			return this.player2;
		}
	}

	public void displayGame() {
		if (!this.winner)
			this.board.displayBoard();
	}

	public ResultAnalyzer getResultAnalyzer() {
		return this.resultAnalyzer;
	}
}

File: dotNet/ClassLibrary/WinForm/TicTacToeApp/TicTacToeCore/Game.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TicTacToeCore
{
    public class Game
    {
        private Player _player1;
        private Player _player2;
        private Board _board;
        private ResultAnalyzer _resultAnalyzer;
        private Player _currentPlayer;
        private bool _turn = true;
        private bool _winner = false;

        public Game(Player player1, Player player2, Board board, ResultAnalyzer resultAnalyzer)
        {
            _player1 = player1;
            _player2 = player2;
            _board = board;
            _resultAnalyzer = resultAnalyzer;
            _currentPlayer = player1;
        }

        public void InitializeGame()
        {
            _board.DisplayBoard();
        }

        public void Play(int position)
        {
            if (ValidChoice(position))
            {
                _currentPlayer = Turn;
                _board.UpdateBoard(CurrentPlayer.Mark, position);

                _resultAnalyzer.AnalyzeBoard();

                if (_turn)
                {
                    _turn = false;
                }
                else
                {
                    _turn = true;
                }
            }
            else
            {
                Console.WriteLine("Choose a valid Option: " + Turn);
            }
        }

        public Player CurrentPlayer
        {
            get
            {
                return _currentPlayer;
            }
        }

        public Board Board
        {
            get
            {
                return _board;
            }
        }

        public Boolean ValidChoice(int position)
        {
            if (!_board.GetCell(position-1).IsMarked())
            {
                return true;
            }
            return false;
        }

        public Player Turn
        {
            get
            {
                if (_turn)
                {
                    return _player1;
                }
                else
                {
                    return _player2;
                }
            }
        }

        public void DisplayGame()
        {
            if (!_winner)
            {
                _board.DisplayBoard();
            }
        }

        public ResultAnalyzer ResultAnalyzer
        {
            get
            {
                return _resultAnalyzer;
            }
        }
    }
}
